packages/txservice/src/provider.ts

```diff
@@ -0,0 +1,177 @@
+import { jsonifyError } from "@connext/nxtp-utils";
+import axios from "axios";
+import { BigNumber, Signer, Wallet, providers } from "ethers";
+import PriorityQueue from "p-queue";
+import { BaseLogger } from "pino";
+
+import { TransactionServiceConfig } from "./config";
+import { ChainError } from "./error";
+import { FullTransaction, MinimalTransaction, ProviderConfig } from "./types";
+
+const { JsonRpcProvider, FallbackProvider } = providers;
+
+/// Could use a more encompassing name, e.g. ChainRpcDispatch, etc
+export class ChainRpcProvider {
+  private provider: providers.JsonRpcProvider | providers.FallbackProvider;
```

Comment:
We should likely prioritize the fallback provider stuff we started in vector much more highly here. How forward looking is this class? Does the contained `provider` cause issues or is that why its an `| providers.FallbackProvider`

May need a refresher on the issues we previously ran into with this

Reply:
previous issues were largely due to the fact that we were using providers all over the place.

No more. All provider instantiation and usage (with the exception of the injected provider in frontend/sdk atm) will be centralized here